scripts/pre_commit_linter.py

```diff
@@ -580,6 +580,94 @@ def _check_bad_pattern_in_file(filename, content, pattern):
     return False
 
 
+def _check_bad_pattern_in_comments_docstrings(filename, content):
+    """Detects whether comments and docstrings end with a period.
+
+    Args:
+        filename: str. Name of the file.
+        content: str. Contents of the file.
+
+    Returns:
+        bool. True if there is bad pattern else false.
+    """
+    regexp_comment = r"#[^\n]*[^.,\n]\n( |\t)*(#(( |\t)*[A-Z])|[^#\s])"
+    message_comment = "There should be a period after comment end."
+    message_docstring = "There should be a period after docstring end."
+    punctuate = ['.',',',';',':']
+    excluded_words = ['utf','pylint:','http://']
```

Comment:
There are some comments like: ``# coding: utf-8`` which need to be excluded while checking since they should not end with a period. I found some of those words and added to this list. We need to update a list whenever any such word is used in a comment.